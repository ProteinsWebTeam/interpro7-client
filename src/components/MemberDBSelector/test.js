// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { _MemberDBSelector as MemberDBSelector } from '.';

const renderer = new ShallowRenderer();

describe('<MemberDBSelector />', () => {
  test('should render', () => {
    renderer.render(
      <MemberDBSelector
        contentType={'entry'}
        lowGraphics={false}
        goToCustomLocation={() => {}}
        customLocation={{
          description: {
            entry: {
              db: 'InterPro',
            },
            main: {
              key: 'entry',
            },
          },
          search: {},
        }}
        dataDB={{
          loading: false,
          payload: {
            databases: {
              interpro: {
                canonical: 'interpro',
                name: 'InterPro',
                releaseDate: '2019-05-09T00:00:00Z',
                type: 'entry',
                version: '74.0',
              },
              cdd: {
                canonical: 'cdd',
                name: 'CDD',
                releaseDate: '2017-03-28T00:00:00Z',
                type: 'entry',
                version: '3.16',
              },
            },
          },
        }}
        dataDBCount={{
          loading: false,
          payload: {
            entries: {
              all: 215965,
              integrated: 49513,
              interpro: 36651,
              unintegrated: 128272,
              member_databases: {
                cdd: 12779,
              },
            },
          },
        }}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
