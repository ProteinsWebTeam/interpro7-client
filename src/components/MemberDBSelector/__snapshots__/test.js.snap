// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemberDBSelector /> should render 1`] = `
<div
  className="container"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyPress={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  role="button"
  tabIndex="0"
>
  <span
    className="child-container"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="0"
  />
  <div
    className="potential-popup"
  >
    <div
      className="filter-label"
    >
      <TooltipAndRTDLink
        rtdPage="searchways.html#using-browse-feature-to-search-and-filter-interpro"
      />
       
      Select your database:
    </div>
    <form
      className="db-selector one-column"
      onChange={[Function]}
    >
      <label
        className="db-choice checked"
        data-testid="memberdb-filter-interpro"
        style={
          Object {
            "color": "#2daec1",
          }
        }
      >
        <input
          checked={true}
          disabled={false}
          name="db-radio"
          onChange={[Function]}
          type="radio"
          value="interpro"
        />
        <span
          className="text"
        >
          InterPro
        </span>
        <Memo(Connect(NumberComponent))
          abbr={true}
          className="label"
          loading={false}
          style={
            Object {
              "background": "#2daec1",
            }
          }
          titleType="entry"
        >
          0
        </Memo(Connect(NumberComponent))>
      </label>
      <label
        className="db-choice"
        data-testid="memberdb-filter-cdd"
        style={
          Object {
            "color": "#addc58",
          }
        }
      >
        <input
          checked={false}
          disabled={false}
          name="db-radio"
          onChange={[Function]}
          type="radio"
          value="cdd"
        />
        <span
          className="text"
        >
          CDD
        </span>
        <Memo(Connect(NumberComponent))
          abbr={true}
          className="label"
          loading={false}
          style={
            Object {
              "background": false,
            }
          }
          titleType="entry"
        >
          0
        </Memo(Connect(NumberComponent))>
      </label>
    </form>
  </div>
</div>
`;
